src/main/webapp/gameover.js

```diff
@@ -1,27 +1,42 @@
 
 // Wait for the page to load then call the startGame function.
-document.addEventListener("DOMContentLoaded", getAndSaveScore);
+document.addEventListener("DOMContentLoaded", displayResults);
 
-function getAndSaveScore() {
-  if (localStorage.getItem("finalScore") == null) {
+// Returns player's final game score and deletes it from local storage
+function getScore() {
+  // Get score from finalscore
+  let score = localStorage.getItem("finalScore");
+  // Clear everything from finalscore to prevent reusing
+  localStorage.removeItem("finalScore");
+  if (score == null) {
     return;
   }
-  let score = Number(localStorage.getItem("finalScore"));
+  return Number(score); 
+}
+
+// Gets player's average score and updates avgScore based on last score
+function getAvg(score) {
   let avgScore = localStorage.getItem("avgScore");
-  if (avgScore == null) {
+
+  // If nothing in avgScore, creates avgScore object and sets game count to 1
+  if (avgScore === null) {
     avgScore = {sum: score, count: 1};
   }
+  // Updates score and number of games in avgScore
   else {
     avgScore = JSON.parse(avgScore);
     avgScore.sum += score;
```

Comment:
i hadn't thought of doing it this way, this is so much simpler than what i had sent in chat!

Reply:
It required some more thinking about it and how to use local storage :^b